src/mds/MDSDaemon.cc

```diff
@@ -832,6 +833,25 @@ void MDSDaemon::handle_mds_map(MMDSMap *m)
   // standby_for_rank if in standby replay)
   mds_rank_t whoami = mdsmap->get_rank_gid(mds_gid_t(monc->get_global_id()));
 
+  if (whoami != MDS_RANK_NONE && mdsmap->is_active(whoami)) {
```

Comment:
This will work, but it would be nicer to put it in MDSRank::handle_mds_map (where whoami != MDS_RANK_NONE is implicitly true)